regression/goto-instrument/value-set-fi-fp-removal/test.c

```diff
@@ -0,0 +1,20 @@
+#include <assert.h>
+
+typedef void (*fp_t)();
+
+void f()
+{
+}
+
+void g()
+{
+}
+
+int main(void)
+{
+  fp_t fp = f;
+  fp();
+
+  // this would fool an analysis that looks for functions whose address is taken
+  fp_t other_fp = g;
```

Comment:
Can we have more expansive tests for this? What about function pointers inside structs, inside arrays, or behind pointers?

Reply:
Have added more tests for all these cases. When the value set is empty it defaults to the behaviour of the standard function pointer removal

Reply:
@hannes-steffenhagen-diffblue I've added more tests, would you have chance to re-review this? Thanks